app/javascript/dashboard/components/NetworkNotification.vue

```diff
@@ -1,26 +1,146 @@
+
+const bannerText = computed(() => {
+  if (isReconnecting.value) return t('NETWORK.NOTIFICATION.RECONNECTING');
+  if (isReconnected.value) return t('NETWORK.NOTIFICATION.RECONNECT_SUCCESS');
+  return t('NETWORK.NOTIFICATION.OFFLINE');
+});
+
+const iconName = computed(() => (isReconnected.value ? 'wifi' : 'wifi-off'));
+const canRefresh = computed(
+  () => !isReconnecting.value && !isReconnected.value
+);
+
+const refreshPage = () => {
+  window.location.reload();
+};
+
+const closeNotification = () => {
+  showNotification.value = false;
+  isReconnected.value = false;
+  clearTimeout(reconnectTimeout);
+};
+
+const isInAnyOfTheRoutes = routeName => {
+  return (
+    isAConversationRoute(routeName, true) ||
+    isAInboxViewRoute(routeName, true) ||
+    isNotificationRoute(routeName, true)
+  );
+};
+
+const updateWebsocketStatus = () => {
+  isDisconnected.value = true;
+  showNotification.value = true;
+};
+
+const handleReconnectionCompleted = () => {
+  isDisconnected.value = false;
+  isReconnecting.value = false;
+  isReconnected.value = true;
+  showNotification.value = true;
+  reconnectTimeout = setTimeout(closeNotification, RECONNECTED_BANNER_TIMEOUT);
+};
+
+const handleReconnecting = () => {
+  if (isInAnyOfTheRoutes(route.name)) {
+    isReconnecting.value = true;
+    isReconnected.value = false;
+    showNotification.value = true;
+  } else {
+    handleReconnectionCompleted();
+  }
+};
+
+const updateOnlineStatus = event => {
+  // Case: Websocket is not disconnected
+  // If the app goes offline, show the notification
+  // If the app goes online, close the notification
+
+  // Case: Websocket is disconnected
+  // If the app goes offline, show the notification
+  // If the app goes online but the websocket is disconnected, don't close the notification
+  // If the app goes online and the websocket is not disconnected, close the notification
+
+  if (event.type === 'offline') {
+    showNotification.value = true;
+  } else if (event.type === 'online' && !isDisconnected.value) {
+    handleReconnectionCompleted();
+  }
+};
+
+const addEventListeners = () => {
+  window.addEventListener('offline', updateOnlineStatus);
+  window.addEventListener('online', updateOnlineStatus);
+  emitter.on(BUS_EVENTS.WEBSOCKET_DISCONNECT, updateWebsocketStatus);
+  emitter.on(
+    BUS_EVENTS.WEBSOCKET_RECONNECT_COMPLETED,
+    handleReconnectionCompleted
+  );
+  emitter.on(BUS_EVENTS.WEBSOCKET_RECONNECT, handleReconnecting);
+};
+
+const removeEventListeners = () => {
+  window.removeEventListener('offline', updateOnlineStatus);
+  window.removeEventListener('online', updateOnlineStatus);
+  emitter.off(BUS_EVENTS.WEBSOCKET_DISCONNECT, updateWebsocketStatus);
+  emitter.off(
+    BUS_EVENTS.WEBSOCKET_RECONNECT_COMPLETED,
+    handleReconnectionCompleted
+  );
+  emitter.off(BUS_EVENTS.WEBSOCKET_RECONNECT, handleReconnecting);
+  clearTimeout(reconnectTimeout);
+};
+
+onMounted(addEventListeners);
+
+onBeforeUnmount(removeEventListeners);
```

Comment:
@iamsivin Found a cleaner way to do this part, the changes were too big to suggest in the review, I've made a small PR https://github.com/chatwoot/chatwoot/pull/9581 with this. Can you please take a look and test it out 

Ref: https://vueuse.org/core/useEventListener/

Reply:
@scmmishra I have tested the PR, And it's working fine.